compiler/rustc_lint/src/autorefs.rs

```diff
@@ -75,10 +75,9 @@ impl<'tcx> LateLintPass<'tcx> for ImplicitAutorefs {
                 _ => return,
             },
             ExprKind::Index(base, _, _) => base,
-            ExprKind::MethodCall(_, inner, _, _)
-                if let Some(def_id) = cx.typeck_results().type_dependent_def_id(expr.hir_id)
-                    && cx.tcx.has_attr(def_id, sym::rustc_no_implicit_autorefs) =>
-            {
+            ExprKind::MethodCall(_, inner, _, _) => {
+                // PERF: Checking of `#[rustc_no_implicit_refs]` is defered below
+                // as checking for attribute is a bit costly.
```

Comment:
```suggestion
                // because checking for attribute is a bit costly.
```
A pet peeve of mine :) "because" is always clear, "as" often isn't because it has more meanings. I had to read this sentence twice to understand it.